src/Jupyter/CustomShell/HeartbeatHandler.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Jupyter.Core;
+using Microsoft.Jupyter.Core.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.Quantum.IQSharp.Jupyter
+{
+
+    public class HeartbeatReplyContent : MessageContent
+    {
+        [JsonProperty("value")]
+        public string Value { get; set; }
+    }
+
+    public class HeartbeatHandler : IShellHandler
```

Comment:
This allows for the JavaScript client to test the two kinds of callback allowed by the `document.IPython.notebook.kernel` object. I think it might make sense to rename from heartbeat to indicate that it is effectively an echo service to help debug client functionality.

Reply:
Refactored to name the heartbeat to an echo and added clarifying comments.